From 15bd58cf692a2d328866ec903ed60e82703ec18c Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Tue, 1 Apr 2014 23:15:39 +0100
Subject: [PATCH] Don't include auto-host.h directly in crtstuff.c

It breaks building glibc on Windows where caddr_t is defined
as "char *" by auto-host.h

libgcc/
	* crtstuff.c: Don't include auto-host.h directly.
---
 libgcc/crtstuff.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/libgcc/crtstuff.c b/libgcc/crtstuff.c
index 12bed4b..3c2df0a 100644
--- a/libgcc/crtstuff.c
+++ b/libgcc/crtstuff.c
@@ -51,7 +51,17 @@ see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
 /* FIXME: Including auto-host is incorrect, but until we have
    identified the set of defines that need to go into auto-target.h,
    this will have to do.  */
-#include "auto-host.h"
+/* Actually don't include this as it breaks building on Windows where
+   caddr_t is defined as "char *" by auto-host.h, which causes glibc's
+   sys/types.h to fall over at "typedef __caddr_t caddr_t;". auto-host.h
+   is included in-directly via tconfig.h and that correctly defines
+   USED_FOR_TARGET first meaning the define of caddr_t isn't done.
+   There is another fix for this:
+   https://sourceware.org/ml/crossgcc/2011-12/msg00077.html
+   .. but until I see this fix causing problems, it seems like the
+   cleanest way. I'm guessing the original FIXME comment has been fixed
+   by using tconfig.h and USED_FOR_TARGET?
+/* #include "auto-host.h" */
 #undef pid_t
 #undef rlim_t
 #undef ssize_t
-- 
2.7.0

